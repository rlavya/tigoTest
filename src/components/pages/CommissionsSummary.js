import { connect } from "react-redux";
import { siteTree } from "../../siteTree";
import { dataForId, configToId } from "../../util/db";
import TransactionsOrCommissionsSummary from "../components/TransactionsOrCommissionsSummary";

export default connect(state => {
  const commissions = dataForId(state, configToId("Commission", "list", true));

  return {
    items: commissions,
    pageName: siteTree.CommissionsSummary.name,
    detailViewPageName: siteTree.CommissionsDetail.name
  };
})(TransactionsOrCommissionsSummary);
