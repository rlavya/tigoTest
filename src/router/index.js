// @flow

import { HELPERS } from "./helpers";
import reduce from "lodash/reduce";

const H = HELPERS;
const QUERY_SEPARATOR = "?";

const URL = root => window.location.hash.substr(root.length);

export type RouteDefinition = {
  defaults: Object,
  length: number,
  names: string[],
  route: string,
  test: RegExp
};

export type RouteData = {
  route?: RouteDefinition,
  [string]: any
};
export type RouteStore = {
  [string]: RouteDefinition
};

const extractQueryParams = (
  url: string,
  start: string,
  parts: string[]
): Object => {
  let remainder = url.substr(
    start.length - (parts[parts.length - 1] === QUERY_SEPARATOR ? 1 : 0)
  );
  if (remainder && H.paramsMatcher.test(remainder)) {
    return H.deparam(remainder.slice(1));
  } else {
    return {};
  }
};

const extractNamesDefaultsAndGenerateRegExp = (
  url,
  defaults
): { names: string[], test: RegExp, defaults: Object } => {
  let names = [];
  let res;
  let test = "";
  let lastIndex = (H.matcher.lastIndex = 0);
  let next;
  let matchSlashes = false;

  defaults = { ...defaults };

  // eslint-disable-next-line
  while ((res = H.matcher.exec(url))) {
    names.push(res[1]);
    test += H.removeBackslash(
      url.substring(lastIndex, H.matcher.lastIndex - res[0].length)
    );
    next =
      "\\" +
      (H.removeBackslash(url.substr(H.matcher.lastIndex, 1)) ||
        QUERY_SEPARATOR + (matchSlashes ? "" : "|/"));
    test += "([^" + next + "]" + (defaults[res[1]] ? "*" : "+") + ")";
    lastIndex = H.matcher.lastIndex;
  }

  test += url.substr(lastIndex).replace("\\", "");

  return {
    names: names,
    defaults: defaults,
    test: new RegExp("^" + test + "($|" + H.wrapQuote(QUERY_SEPARATOR) + ")")
  };
};

const generateURLfromRoute = (
  route: RouteDefinition,
  data: { [string]: any }
): string => {
  let cpy = { ...data };
  let res = route.route
    .replace(H.matcher, function(whole, name) {
      delete cpy[name];
      return data[name] === route.defaults[name]
        ? ""
        : encodeURIComponent(data[name]);
    })
    .replace("\\", "");

  for (let name in route.defaults) {
    if (route.defaults.hasOwnProperty(name)) {
      let val = route.defaults[name];
      if (cpy[name] === val) {
        delete cpy[name];
      }
    }
  }

  let after = H.param(cpy);

  return res + (after ? "?" + after : "");
};

export class Router {
  _routes: RouteStore;
  _responders: Function[];
  _store: Object;
  _prevRouteData: RouteData;
  routeData: RouteData;
  root: string;
  stop: Function;

  constructor(store: Object, routes: Object[], responders: Function[]) {
    this._routes = {};
    this._store = store;
    this._responders = responders || [];

    routes.forEach(args => {
      let [route, defaults] = args;
      this.add(route, defaults);
    });

    this.root = "#!";
  }

  start(): void {
    let hashChangedHandler = _ => {
      this._prevRouteData = this.routeData;
      this.setRouteData(URL(this.root));
      this.dispatch();
      this._responders.forEach(r => {
        r(this._store, this.routeData, this._prevRouteData);
      });
    };

    this.stop = () => {
      window.removeEventListener("hashchange", hashChangedHandler);
    };

    window.addEventListener("hashchange", hashChangedHandler);
    hashChangedHandler();
  }

  restartResponders(): void {
    this._responders.forEach(r => {
      r(this._store, this.routeData, this._prevRouteData, true);
    });
  }

  dispatch(): void {
    this._store.dispatch({ type: "ROUTE_CHANGED", payload: this.routeData });
  }

  dataFor(data: Object, merge: boolean): Object {
    if (merge) {
      let routeData = { ...this.routeData };
      delete routeData.route;

      return {
        ...routeData,
        ...data
      };
    }

    return { ...data };
  }

  urlFor(data: Object, merge: boolean): string {
    return this.root + this.param(this.dataFor(data, merge));
  }

  setRouteData(url: string): void {
    this.routeData = this.deparam(url);
  }

  get(attr: string): any {
    let data = this.routeData;
    if (attr) {
      return data[attr];
    }
    return data;
  }

  param(data: Object): string {
    let routeName = data.route;
    let route: RouteDefinition;
    let matches: number;

    delete data.route;

    ({ route, matches } = reduce(
      this._routes,
      (acc, r) => {
        const matchCount = H.matchesData(r, data);
        if (matchCount > acc.matches) {
          return {
            matches: matchCount,
            route: r
          };
        }
        return acc;
      },
      {
        matches: 0,
        route: null
      }
    ));

    if (
      this._routes[routeName] &&
      H.matchesData(this._routes[routeName], data) === matches
    ) {
      route = this._routes[routeName];
    }

    if (route) {
      return generateURLfromRoute(route, data);
    }
    return Object.keys(data).length === 0 ? "" : "?" + H.param(data);
  }

  deparam(url: string): RouteData {
    const route: RouteDefinition = reduce(
      this._routes,
      (acc, r) => {
        let match = r.test.test(url);
        if (match && r.length > acc.length) {
          return r;
        }
        return acc;
      },
      { length: -1 }
    );

    if (route) {
      let parts: string[] = url.match(route.test) || [];
      let start = parts.shift();

      let routeData = {
        ...route.defaults,
        ...extractQueryParams(url, start, parts),
        route
      };

      return reduce(
        parts,
        (acc, part, i) => {
          if (part && part !== QUERY_SEPARATOR) {
            acc[route.names[i]] = decodeURIComponent(part);
          }
          return acc;
        },
        routeData
      );
    }
    if (url.charAt(0) !== QUERY_SEPARATOR) {
      url = QUERY_SEPARATOR + url;
    }
    return H.paramsMatcher.test(url) ? H.deparam(url.slice(1)) : {};
  }

  add(url: string, defaults: Object): void {
    if (url.indexOf("/") === 0) {
      url = url.substr(1);
    }

    let route: RouteDefinition = {
      length: url.split("/").length,
      route: url,
      ...extractNamesDefaultsAndGenerateRegExp(url, defaults || {})
    };

    this._routes[url] = route;
  }

  clear(): void {
    this._routes = {};
  }
}

let __ROUTER_CACHE__ = null;

export function startRouter(
  store: Object,
  routes: Object[],
  responders: Function[]
) {
  if (__ROUTER_CACHE__) {
    __ROUTER_CACHE__.stop();
  }

  __ROUTER_CACHE__ = new Router(store, routes, responders);
  __ROUTER_CACHE__.start();

  return __ROUTER_CACHE__;
}

export function urlFor(data: Object, merge: boolean) {
  if (!__ROUTER_CACHE__) {
    throw new Error("Router must be started!");
  }

  return __ROUTER_CACHE__.urlFor(data, merge);
}

export function restartResponders() {
  if (!__ROUTER_CACHE__) {
    throw new Error("Router must be started!");
  }

  return __ROUTER_CACHE__.restartResponders();
}

export function redirectTo(data: Object, merge: boolean) {
  window.location.hash = urlFor(data, merge);
}
