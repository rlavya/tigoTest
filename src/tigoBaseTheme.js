const tigoBaseTheme = {
  fontFamily: "Roboto, sans-serif",
  borderRadius: 2,
  palette: {
    primary1Color: "#00377B",
    primary2Color: "#4CC5F2",
    primary3Color: "#333",
    accent1Color: "#4CC5F2",
    accent2Color: "#4CC5F2",
    accent3Color: "#4CC5F2",
    textColor: "#00377B",
    secondaryTextColor: "#839ebf",
    alternateTextColor: "#fafafa",
    canvasColor: "#FFFFFF"
  }
};

export default tigoBaseTheme;
