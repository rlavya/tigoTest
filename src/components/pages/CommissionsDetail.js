import { connect } from "react-redux";
import { siteTree } from "../../siteTree";
import { dataForId, configToId } from "../../util/db";
import { redirectTo } from "../../router";
import TransactionsOrCommissionsDetail from "../components/TransactionsOrCommissionsDetail";

export default connect(
  state => {
    const items = dataForId(
      state,
      configToId("CommissionDetail", "list", true)
    );
    return {
      items,
      pageName: siteTree.TransactionsDetail.name,
      periodFilter: state.route.periodFilter || "hora",
      transactionFilter: state.route.transactionFilter || "todas"
    };
  },
  dispatch => {
    return {
      changePeriodFilter(filter) {
        redirectTo({ periodFilter: filter }, true);
      },
      changeTransactionFilter(filter) {
        redirectTo({ transactionFilter: filter }, true);
      }
    };
  }
)(TransactionsOrCommissionsDetail);
