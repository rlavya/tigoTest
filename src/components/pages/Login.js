import React, { Component } from "react";
import RaisedButton from "material-ui/RaisedButton";
import TextField from "material-ui/TextField";
import PageTitle from "../components/PageTitle";
import { lightBlue200, indigo300 } from "material-ui/styles/colors";
import get from "lodash/get";
import { connect } from "react-redux";
import {
  updatePosPinAction,
  updatePosDMSIDAction,
  loginAction
} from "../../actionCreators/auth.js";
import styled from "styled-components";
import { urlFor } from "../../router";
import { onEnterAction } from "../../util";

const InputWrap = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  img {
    display: block;
    margin-bottom: -23px;
    margin-right: 5px;
    margin-left: -15px;
  }
`;

export class Login extends Component {
  render() {
    const loginIfValid = onEnterAction(() => {
      if (this.props.pin.trim() !== "" && this.props.dmsid.trim() !== "") {
        this.props.login();
      }
    });

    return (
      <PageTitle page="Login">
        <div className="max-width-4 mx-auto">
          <div className="clearfix center">
            <h1 className="light text-primary h2">Ingresa tus credenciales:</h1>
            <InputWrap>
              <img src="/images/icn-DMSID.png" alt="" />
              <TextField
                underlineFocusStyle={{ borderColor: lightBlue200 }}
                floatingLabelStyle={{ color: indigo300 }}
                floatingLabelText="DMS ID"
                id="dmsid"
                value={this.props.dmsid}
                onChange={this.props.updateDMSID}
                type="tel"
                pattern="[0-9]*"
                onKeyDown={loginIfValid}
                inputStyle={{ fontSize: "1.5rem" }}
              />
            </InputWrap>
            <InputWrap>
              <img src="/images/icn-password.png" alt="" />
              <div className="masked-input">
                <TextField
                  underlineFocusStyle={{ borderColor: lightBlue200 }}
                  floatingLabelStyle={{ color: indigo300 }}
                  floatingLabelText="PIN"
                  id="password"
                  type="tel"
                  pattern="[0-9]*"
                  value={this.props.pin}
                  onChange={this.props.updatePin}
                  onKeyDown={loginIfValid}
                  inputStyle={{ fontSize: "1.5rem" }}
                />
              </div>
            </InputWrap>
            <br />

            <RaisedButton
              className="mt2"
              secondary={true}
              label="Iniciar"
              onClick={this.props.login}
              disabledBackgroundColor="#CAF1F6"
              disabledLabelColor="#FFF"
              disabled={
                this.props.pin.trim() === "" || this.props.dmsid.trim() === ""
              }
            />
            <br />
            <p>
              <a
                href={urlFor({ page: "ChangePin" })}
                className="text-secondary text-decoration-none light"
              >
                Cambiar PIN
              </a>
            </p>
          </div>
        </div>
      </PageTitle>
    );
  }
}

export default connect(
  state => {
    return {
      dmsid: get(state, "auth.dmsid") || "",
      pin: get(state, "auth.pin") || ""
    };
  },
  dispatch => {
    return {
      updatePin: (e, value) => {
        return dispatch(updatePosPinAction(value));
      },
      updateDMSID: (e, value) => {
        return dispatch(updatePosDMSIDAction(value));
      },
      login: (e, value) => {
        return dispatch(loginAction());
      }
    };
  }
)(Login);
