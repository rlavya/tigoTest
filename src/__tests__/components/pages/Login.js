import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import getMuiTheme from "material-ui/styles/getMuiTheme";
import tigoBaseTheme from "../../../tigoBaseTheme";
import { Login } from "../../../components/pages/Login";
import renderer from "react-test-renderer";
import Enzyme, { shallow, render, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import React from "react";

import { startRouter } from "../../../router";

Enzyme.configure({ adapter: new Adapter() });


let mockStore = {
  dispatch(action) {},
};

startRouter(mockStore, [[":page", { page: "MainMenu" }]], []);

describe('Login component', () => {

  const urlFor = jest.fn();

  it("renders correctly", () => {
    const component = shallow(
      <Login dmsid="" pin=""/>
    );

    expect(component).toMatchSnapshot();
  });

});
