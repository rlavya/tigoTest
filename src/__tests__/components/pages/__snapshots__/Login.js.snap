// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    dmsid=""
    pin=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(PageTitle)
          page="Login"
/>,
        <div
          className="clearfix center"
>
          <h1
                    className="light text-primary h2"
          >
                    Ingresa tus credenciales:
          </h1>
          <styled.div>
                    <img
                              alt=""
                              src="/images/icn-DMSID.png"
                    />
                    <TextField
                              disabled={false}
                              floatingLabelFixed={false}
                              floatingLabelStyle={
                                        Object {
                                                  "color": "#7986cb",
                                                }
                              }
                              floatingLabelText="DMSID"
                              fullWidth={false}
                              id="dmsid"
                              multiLine={false}
                              onChange={undefined}
                              rows={1}
                              type="text"
                              underlineFocusStyle={
                                        Object {
                                                  "borderColor": "#81d4fa",
                                                }
                              }
                              underlineShow={true}
                              value=""
                    />
          </styled.div>
          <styled.div>
                    <img
                              alt=""
                              src="/images/icn-password.png"
                    />
                    <TextField
                              disabled={false}
                              floatingLabelFixed={false}
                              floatingLabelStyle={
                                        Object {
                                                  "color": "#7986cb",
                                                }
                              }
                              floatingLabelText="CLAVE"
                              fullWidth={false}
                              id="password"
                              multiLine={false}
                              onChange={undefined}
                              rows={1}
                              type="password"
                              underlineFocusStyle={
                                        Object {
                                                  "borderColor": "#81d4fa",
                                                }
                              }
                              underlineShow={true}
                              value=""
                    />
          </styled.div>
          <br />
          <RaisedButton
                    className="mt2"
                    disabled={true}
                    disabledBackgroundColor="#CAF1F6"
                    disabledLabelColor="#FFF"
                    fullWidth={false}
                    label="Iniciar"
                    labelPosition="after"
                    onClick={undefined}
                    primary={false}
                    secondary={true}
          />
          <br />
          <p>
                    <a
                              className="text-secondary text-decoration-none light"
                              href="#!ChangePin"
                    >
                              Cambiar Clave
                    </a>
          </p>
</div>,
      ],
      "className": "max-width-4 mx-auto",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "page": "Login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="light text-primary h2"
>
              Ingresa tus credenciales:
</h1>,
            <styled.div>
              <img
                            alt=""
                            src="/images/icn-DMSID.png"
              />
              <TextField
                            disabled={false}
                            floatingLabelFixed={false}
                            floatingLabelStyle={
                                          Object {
                                                        "color": "#7986cb",
                                                      }
                            }
                            floatingLabelText="DMSID"
                            fullWidth={false}
                            id="dmsid"
                            multiLine={false}
                            onChange={undefined}
                            rows={1}
                            type="text"
                            underlineFocusStyle={
                                          Object {
                                                        "borderColor": "#81d4fa",
                                                      }
                            }
                            underlineShow={true}
                            value=""
              />
</styled.div>,
            <styled.div>
              <img
                            alt=""
                            src="/images/icn-password.png"
              />
              <TextField
                            disabled={false}
                            floatingLabelFixed={false}
                            floatingLabelStyle={
                                          Object {
                                                        "color": "#7986cb",
                                                      }
                            }
                            floatingLabelText="CLAVE"
                            fullWidth={false}
                            id="password"
                            multiLine={false}
                            onChange={undefined}
                            rows={1}
                            type="password"
                            underlineFocusStyle={
                                          Object {
                                                        "borderColor": "#81d4fa",
                                                      }
                            }
                            underlineShow={true}
                            value=""
              />
</styled.div>,
            <br />,
            <RaisedButton
              className="mt2"
              disabled={true}
              disabledBackgroundColor="#CAF1F6"
              disabledLabelColor="#FFF"
              fullWidth={false}
              label="Iniciar"
              labelPosition="after"
              onClick={undefined}
              primary={false}
              secondary={true}
/>,
            <br />,
            <p>
              <a
                            className="text-secondary text-decoration-none light"
                            href="#!ChangePin"
              >
                            Cambiar Clave
              </a>
</p>,
          ],
          "className": "clearfix center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ingresa tus credenciales:",
              "className": "light text-primary h2",
            },
            "ref": null,
            "rendered": "Ingresa tus credenciales:",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <img
                  alt=""
                  src="/images/icn-DMSID.png"
/>,
                <TextField
                  disabled={false}
                  floatingLabelFixed={false}
                  floatingLabelStyle={
                                    Object {
                                                      "color": "#7986cb",
                                                    }
                  }
                  floatingLabelText="DMSID"
                  fullWidth={false}
                  id="dmsid"
                  multiLine={false}
                  onChange={undefined}
                  rows={1}
                  type="text"
                  underlineFocusStyle={
                                    Object {
                                                      "borderColor": "#81d4fa",
                                                    }
                  }
                  underlineShow={true}
                  value=""
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "/images/icn-DMSID.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "disabled": false,
                  "floatingLabelFixed": false,
                  "floatingLabelStyle": Object {
                    "color": "#7986cb",
                  },
                  "floatingLabelText": "DMSID",
                  "fullWidth": false,
                  "id": "dmsid",
                  "multiLine": false,
                  "onChange": undefined,
                  "rows": 1,
                  "type": "text",
                  "underlineFocusStyle": Object {
                    "borderColor": "#81d4fa",
                  },
                  "underlineShow": true,
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <img
                  alt=""
                  src="/images/icn-password.png"
/>,
                <TextField
                  disabled={false}
                  floatingLabelFixed={false}
                  floatingLabelStyle={
                                    Object {
                                                      "color": "#7986cb",
                                                    }
                  }
                  floatingLabelText="CLAVE"
                  fullWidth={false}
                  id="password"
                  multiLine={false}
                  onChange={undefined}
                  rows={1}
                  type="password"
                  underlineFocusStyle={
                                    Object {
                                                      "borderColor": "#81d4fa",
                                                    }
                  }
                  underlineShow={true}
                  value=""
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "/images/icn-password.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "disabled": false,
                  "floatingLabelFixed": false,
                  "floatingLabelStyle": Object {
                    "color": "#7986cb",
                  },
                  "floatingLabelText": "CLAVE",
                  "fullWidth": false,
                  "id": "password",
                  "multiLine": false,
                  "onChange": undefined,
                  "rows": 1,
                  "type": "password",
                  "underlineFocusStyle": Object {
                    "borderColor": "#81d4fa",
                  },
                  "underlineShow": true,
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "mt2",
              "disabled": true,
              "disabledBackgroundColor": "#CAF1F6",
              "disabledLabelColor": "#FFF",
              "fullWidth": false,
              "label": "Iniciar",
              "labelPosition": "after",
              "onClick": undefined,
              "primary": false,
              "secondary": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="text-secondary text-decoration-none light"
                href="#!ChangePin"
>
                Cambiar Clave
</a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cambiar Clave",
                "className": "text-secondary text-decoration-none light",
                "href": "#!ChangePin",
              },
              "ref": null,
              "rendered": "Cambiar Clave",
              "type": "a",
            },
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(PageTitle)
            page="Login"
/>,
          <div
            className="clearfix center"
>
            <h1
                        className="light text-primary h2"
            >
                        Ingresa tus credenciales:
            </h1>
            <styled.div>
                        <img
                                    alt=""
                                    src="/images/icn-DMSID.png"
                        />
                        <TextField
                                    disabled={false}
                                    floatingLabelFixed={false}
                                    floatingLabelStyle={
                                                Object {
                                                            "color": "#7986cb",
                                                          }
                                    }
                                    floatingLabelText="DMSID"
                                    fullWidth={false}
                                    id="dmsid"
                                    multiLine={false}
                                    onChange={undefined}
                                    rows={1}
                                    type="text"
                                    underlineFocusStyle={
                                                Object {
                                                            "borderColor": "#81d4fa",
                                                          }
                                    }
                                    underlineShow={true}
                                    value=""
                        />
            </styled.div>
            <styled.div>
                        <img
                                    alt=""
                                    src="/images/icn-password.png"
                        />
                        <TextField
                                    disabled={false}
                                    floatingLabelFixed={false}
                                    floatingLabelStyle={
                                                Object {
                                                            "color": "#7986cb",
                                                          }
                                    }
                                    floatingLabelText="CLAVE"
                                    fullWidth={false}
                                    id="password"
                                    multiLine={false}
                                    onChange={undefined}
                                    rows={1}
                                    type="password"
                                    underlineFocusStyle={
                                                Object {
                                                            "borderColor": "#81d4fa",
                                                          }
                                    }
                                    underlineShow={true}
                                    value=""
                        />
            </styled.div>
            <br />
            <RaisedButton
                        className="mt2"
                        disabled={true}
                        disabledBackgroundColor="#CAF1F6"
                        disabledLabelColor="#FFF"
                        fullWidth={false}
                        label="Iniciar"
                        labelPosition="after"
                        onClick={undefined}
                        primary={false}
                        secondary={true}
            />
            <br />
            <p>
                        <a
                                    className="text-secondary text-decoration-none light"
                                    href="#!ChangePin"
                        >
                                    Cambiar Clave
                        </a>
            </p>
</div>,
        ],
        "className": "max-width-4 mx-auto",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "page": "Login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="light text-primary h2"
>
                Ingresa tus credenciales:
</h1>,
              <styled.div>
                <img
                                alt=""
                                src="/images/icn-DMSID.png"
                />
                <TextField
                                disabled={false}
                                floatingLabelFixed={false}
                                floatingLabelStyle={
                                                Object {
                                                                "color": "#7986cb",
                                                              }
                                }
                                floatingLabelText="DMSID"
                                fullWidth={false}
                                id="dmsid"
                                multiLine={false}
                                onChange={undefined}
                                rows={1}
                                type="text"
                                underlineFocusStyle={
                                                Object {
                                                                "borderColor": "#81d4fa",
                                                              }
                                }
                                underlineShow={true}
                                value=""
                />
</styled.div>,
              <styled.div>
                <img
                                alt=""
                                src="/images/icn-password.png"
                />
                <TextField
                                disabled={false}
                                floatingLabelFixed={false}
                                floatingLabelStyle={
                                                Object {
                                                                "color": "#7986cb",
                                                              }
                                }
                                floatingLabelText="CLAVE"
                                fullWidth={false}
                                id="password"
                                multiLine={false}
                                onChange={undefined}
                                rows={1}
                                type="password"
                                underlineFocusStyle={
                                                Object {
                                                                "borderColor": "#81d4fa",
                                                              }
                                }
                                underlineShow={true}
                                value=""
                />
</styled.div>,
              <br />,
              <RaisedButton
                className="mt2"
                disabled={true}
                disabledBackgroundColor="#CAF1F6"
                disabledLabelColor="#FFF"
                fullWidth={false}
                label="Iniciar"
                labelPosition="after"
                onClick={undefined}
                primary={false}
                secondary={true}
/>,
              <br />,
              <p>
                <a
                                className="text-secondary text-decoration-none light"
                                href="#!ChangePin"
                >
                                Cambiar Clave
                </a>
</p>,
            ],
            "className": "clearfix center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ingresa tus credenciales:",
                "className": "light text-primary h2",
              },
              "ref": null,
              "rendered": "Ingresa tus credenciales:",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <img
                    alt=""
                    src="/images/icn-DMSID.png"
/>,
                  <TextField
                    disabled={false}
                    floatingLabelFixed={false}
                    floatingLabelStyle={
                                        Object {
                                                            "color": "#7986cb",
                                                          }
                    }
                    floatingLabelText="DMSID"
                    fullWidth={false}
                    id="dmsid"
                    multiLine={false}
                    onChange={undefined}
                    rows={1}
                    type="text"
                    underlineFocusStyle={
                                        Object {
                                                            "borderColor": "#81d4fa",
                                                          }
                    }
                    underlineShow={true}
                    value=""
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "src": "/images/icn-DMSID.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "disabled": false,
                    "floatingLabelFixed": false,
                    "floatingLabelStyle": Object {
                      "color": "#7986cb",
                    },
                    "floatingLabelText": "DMSID",
                    "fullWidth": false,
                    "id": "dmsid",
                    "multiLine": false,
                    "onChange": undefined,
                    "rows": 1,
                    "type": "text",
                    "underlineFocusStyle": Object {
                      "borderColor": "#81d4fa",
                    },
                    "underlineShow": true,
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <img
                    alt=""
                    src="/images/icn-password.png"
/>,
                  <TextField
                    disabled={false}
                    floatingLabelFixed={false}
                    floatingLabelStyle={
                                        Object {
                                                            "color": "#7986cb",
                                                          }
                    }
                    floatingLabelText="CLAVE"
                    fullWidth={false}
                    id="password"
                    multiLine={false}
                    onChange={undefined}
                    rows={1}
                    type="password"
                    underlineFocusStyle={
                                        Object {
                                                            "borderColor": "#81d4fa",
                                                          }
                    }
                    underlineShow={true}
                    value=""
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "src": "/images/icn-password.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "disabled": false,
                    "floatingLabelFixed": false,
                    "floatingLabelStyle": Object {
                      "color": "#7986cb",
                    },
                    "floatingLabelText": "CLAVE",
                    "fullWidth": false,
                    "id": "password",
                    "multiLine": false,
                    "onChange": undefined,
                    "rows": 1,
                    "type": "password",
                    "underlineFocusStyle": Object {
                      "borderColor": "#81d4fa",
                    },
                    "underlineShow": true,
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "mt2",
                "disabled": true,
                "disabledBackgroundColor": "#CAF1F6",
                "disabledLabelColor": "#FFF",
                "fullWidth": false,
                "label": "Iniciar",
                "labelPosition": "after",
                "onClick": undefined,
                "primary": false,
                "secondary": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="text-secondary text-decoration-none light"
                  href="#!ChangePin"
>
                  Cambiar Clave
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Cambiar Clave",
                  "className": "text-secondary text-decoration-none light",
                  "href": "#!ChangePin",
                },
                "ref": null,
                "rendered": "Cambiar Clave",
                "type": "a",
              },
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
